### ISSUE TITLE
Using Primary64 in model mapping and bigint

### ISSUE THREAD

**Issue by DmytroStepaniuk** [imdrasil/jennifer.cr#142]
Hi! 

Since I've got `bigint` for `id` column in `users` table

Im trying to use `User` model like that: 

```crystal
class User < ApplicationRecord
  with_timestamps

  mapping(
    id:                     Primary64, # same as {type: Int32, primary: true}
    email:                  {type: String?},
    created_at:             {type: Time, null: false, default: Time.now},
    updated_at:             {type: Time, null: false, default: Time.now}
  )
end

```

Then

When launch the app

Im getting errors like that:

```
expanding macro
in macro '__field_declaration' /Users/admin/prj/crystal/amber_backend/lib/jennifer/src/jennifer/model/mapping.cr:18, line 53:

...

  50.               # Inits primary field
  51.               def init_primary_field(value)
  52.                 raise ::Jennifer::AlreadyInitialized.new(@id, value) if @id
> 53.                 @id = value.as(Primary64)
  54.               end
  55.

...

can't cast Int32 to Int64
03:48:16 Watcher    | (INFO) Compile time errors detected. Shutting down...
```

Whats wrong with it and why Im not able to use `Primary64` in this case? 🤔 

When I was using `Primary32` I got:

```
2018-06-15 15:49:52 +03:00: TRANSACTION START
2018-06-15 15:49:52 +03:00: 594 µs INSERT INTO users(email, created_at, updated_at) VALUES ($1, $2, $3)  RETURNING id | ["some@exmaple.com", 2018-06-15 12:49:52 UTC, 2018-06-15 12:49:52 UTC]
2018-06-15 15:49:52 +03:00: TRANSACTION ROLLBACK
```

```
ERROR: cast from Int64 to Int32 failed
```

*UPD*

`integer` works perfectly with `Int32` mapping type 🤔 

```sql
\d+ users
                                               Table "public.users"
   Column   |            Type             | Collation | Nullable | Default | Storage | Stats target | Description
------------+-----------------------------+-----------+----------+---------+---------+--------------+-------------
 created_at | timestamp without time zone |           |          |         | plain   |              |
 updated_at | timestamp without time zone |           |          |         | plain   |              |
 id         | integer                     |           |          |         | plain   |              |
```

but not for `bigint`

```sql
\d+ users
                                                                  Table "public.users"
        Column         |            Type             | Collation | Nullable |              Default              | Storage  | Stats target | Description
-----------------------+-----------------------------+-----------+----------+-----------------------------------+----------+--------------+-------------
 id                    | bigint                      |           | not null | nextval('users_id_seq'::regclass) | plain    |              |
```

**Comment by imdrasil** [comment]
Yeah, this is know issue (unfortunately) on the master branch. This is fixed on the `v0.6.0` branch, which also includes a lot of changes regarding introducing integrating with `amber` framework. Unfortunately, I will be able to merge and release it only after resolving all integrating issues. If you will face the same issue on `v0.6.0` keep posting here.

P.S. `Primary64` actually same as `{type: Int64, primary: true}`

**Comment by imdrasil** [comment]
As jennifer v0.6.0 is release I believe this issue should be considered as fixed. If you still face the same behavior - feel free to reopen this one.
